### ISSUE TITLE
cmd/upspin: cp error reported too lazily for huge files

### ISSUE THREAD

**Issue by masiulaniec** [upspin/upspin#473]
The following command:

    upspin cp HugeFile @/path/that/does/not/exist

... will waste resources copying the blocks to the storeserver only to get aborted at the very end by the dirserver due to a path error. Perhaps the error condition could be discovered earlier?

**Comment by adg** [comment]
Agreed. cp should do a WhichAccess and check that we have write access to the destination path before writing any blocks to storage.

**Comment by PierreZ** [comment]
Hi! If there's no objection, I would like to give it a try as my first PR to Upspin :smiley:

**Comment by adg** [comment]
@PierreZ your help would be most appreciated! Although note that we don't do PRs; we use a code review system called gerrit instead. See: https://github.com/upspin/upspin/#contributing

**Comment by PierreZ** [comment]
Sorry for my misuse of the PR term, I updated my comment.

**Comment by adg** [comment]
No worries, just didn't want you to do unnecessary work! :-)

**Comment by robpike** [comment]
I think that `upspin cp` is not the right location for this check, as `upspin cp` is a relatively minor player in the creation of Upspin storage, or at least I hope that is or soon will be the case.

This should be a check in the client library, and one made efficient by careful caching of WhichAccess call results coupled to attention paid to updated Access and Group files.
